### DISCUSSION TITLE
Provide pkgconfig via the CMake toolchain file on Windows

### DISCUSSION THREAD

**Discussion by glebm** [microsoft/vcpkg#19190] [Ideas]
**Is your feature request related to a problem? Please describe.**

The Windows experience would be much nicer if vcpkg CMake toolchain file provided pkgconfig and set `PKG_CONFIG_EXECUTABLE`.

**Proposed solution**

Set PKG_CONFIG CMake variables in the CMake toolchain file.

**Additional context**

Currently users have to work around this by downloading and building pkgconfig themselves.
Internally, vcpkg already seems to have much of what would be needed to support it.

**Comment by JackBoosY** [comment]
We do need this.
cc @Neumann-A

**Comment by Neumann-A** [comment]
hmm added #17488 and #17487 so people can choose which one they want.

**Comment by JackBoosY** [comment]
@Neumann-A Also, we should output the usage of pkgconfig I think.
